qabelbox/src/main/java/de/qabel/qabelbox/adapter/QabelSyncAdapter.java

```diff
@@ -52,21 +66,48 @@ public void onPerformSync(
 		    SyncResult syncResult) {
         Log.w(TAG, "Starting drop message sync");
         ChatServer chatServer = new ChatServer(context);
-        for (Identity identity: getIdentities()) {
+        Set<Identity> identities;
+        try {
+            identities = getIdentities().getIdentities();
+        } catch (SQLException | EntityNotFoundExcepion | PersistenceException e) {
+            Log.e(TAG, "Sync failed", e);
+            return;
+        }
+        for (Identity identity: identities) {
             Log.i(TAG, "Loading messages for identity "+ identity.getAlias());
-            chatServer.refreshList(getDropConnector(), identity);
+            try {
+                chatServer.refreshList(getDropConnector(), identity);
+            } catch (SQLException | EntityNotFoundExcepion | PersistenceException e) {
+                Log.e(TAG, "Drop message retrieval failed", e);
+                return;
+            }
         }
+        sendRefreshIntents();
+    }
+
+    private void sendRefreshIntents() {
         Intent intent = new Intent(Helper.INTENT_REFRESH_CONTACTLIST);
         context.sendBroadcast(intent);
         Intent chatIntent = new Intent(Helper.INTENT_REFRESH_CHAT);
         context.sendBroadcast(chatIntent);
     }
 
-    private List<Identity> getIdentities() {
-        return new ArrayList<>();
+    private Identities getIdentities() throws SQLException, EntityNotFoundExcepion, PersistenceException {
+        AndroidClientDatabase database = factory.getAndroidClientDatabase();
+        return factory.getIdentityRepository(database).findAll();
```

Comment:
The Interface sounds redundant. The factory already knows the database so you could just call a `getIdentityRepository()`